File: hw2/sessionpage1.pl
#!/usr/bin/perl

use Data::Dumper;
use CGI qw/:standard :html3/;
use CGI::Carp qw(fatalsToBrowser); # show errors in browser
use CGI::Session;

my $q = CGI->new;
 
my $s = CGI::Session->new($q);

print $s->header();

print $q->header;




$newStyle=<<END;
<!-- 

Body {
    background-color: #CCDDEE
}
P.Tip {
margin-right: 50pt;
margin-left: 50pt;
    color: red;
}
P.Alert {
font-size: 30pt;
    font-family: sans-serif;
  color: red;
}
-->
END

print $q->start_html(-title=>'Insert Form CGI',
    -style=>{
        -code=>$newStyle
    }
);


print "<script src='form.js'    type='text/javascript'></script>\n";


printf "Your session ID is: %s\n", $s->id;
printf "This sessin is: %s\n", $s->is_new ? 'NEW': 'old';
printf "Stored session 'test' value: '%s'\n", $q->escapeHTML($s->param('test'));
printf "CGI Params: %s\n", join ', ', $q->param;


print $q->start_form(
        -name    => 'main_form',
        -id      => 'myForm',
        -method  => 'POST',
        -enctype => &CGI::URL_ENCODED,
        -onsubmit => 'validate_session()',
        -action => 'sessionpage2.pl', # Defaults to 
                                                 # the current program
);


print "Name: ".$q->textfield(
        -name      => 'name',
        -value     => '',
        -size      => 20,
        -maxlength => 30,
    );

# print $q->br;

# print "Favorite Color: ".$q->textfield(
#         -name      => 'color',
#         -value     => '',
#         -size      => 20,
#         -maxlength => 30,
#     );

print $q->br;

# print $q->popup_menu(-name=>'ins_method',
# 				 -values=>['GET','POST'],
# 				 -default=>'POST');

# print $q->br;

print $q->submit(-value=>'Submit');

print $q->end_form;

print $q->end_html;

File: hw2/sessionpage1.php
<!DOCTYPE HTML>
<html lang="en-US">
<head>
</head>

<body>

<script src='form.js'    type='text/javascript'></script>
<?php
	// echo '<p>Hello World from PHP on ' . date('Y-m-d h:i:s a') . '</p>';

	session_start();

	$background_color = '#CCDDEE';

	echo "<body style='background-color:$background_color'>";


?>




<form id='myForm' name='main_form' method='POST' onsubmit='validate_session()' action ='echo'>

	<p>Name: <input type="text" name="name" value="<?php sessionpage2 $name;?>"></p>
	<!-- <p>Favorite Color: <input type="text" name="color" value="<?php echo $color;?>"></p> -->


<!-- 	<select name="ins_method">
		<option value="POST">POST</option>
		<option value="GET">GET</option>
	</select>
 -->
	<input type="submit" value="Submit">

</form>



</body>

</html>